### ISSUE TITLE
Scrolling beyond the map boundaries

### ISSUE THREAD

**Issue by ghost** [mapbox/DEPRECATED-mapbox-ios-sdk#473]
Hi there :)

Is there an option to create a MapBox-view like in the image attached?
The free space should be part of the TileView, it shouldn't be just a container containing the full size tileView.

![293d403c-1333-11e4-8838-b0121c7ca136](https://cloud.githubusercontent.com/assets/8256996/3720700/8b082354-1658-11e4-8ba8-6ece566e47e5.png)

Thank you very much! Mapbox and its iOS lib are great :+1:

**Comment by incanus** [comment]
I'm not sure why you wouldn't just apply a mask view above the map view in the view hierarchy? Can you describe the use case some more?

**Comment by ghost** [comment]
Thank you for your fast reply.
There is an virtual crosshair in my app to select a point on the map in detail.
It should be possible to select the corners of the map too while using zoom level 0 :)

**Comment by incanus** [comment]
Ah, I think I understand. 

Ok, the thing about `z0` is that the entire world is one `256px` tile. So the map view cannot be larger than that. If you create a `256px` square map view, you should be able to set `z0` and accomplish your goal.

**Comment by ghost** [comment]
I am not sure I understand your answer and I suspect that is because I didn't make myself entirely clear about what I am trying to accomplish. Let me rephrase my problem:

# Definitions

I will use the term route-me library to indicate all forks of this code base.

# Goal

Enable the selection of arbitrary map locations via a crosshair placed in the center of device screen.

# Problem

The route-me library does not allow scrolling beyond the map boundaries, which would be needed in order to select a location at the very edge of a map.

# Questions

Which exact parts of the route-me library would have to be modified as to allow scrolling beyond the map boundaries?

**Comment by incanus** [comment]
Ah, I understand now. I was confused by the partial-world nature of the map you showed. But ideally, you would want the user to be able to select, say, the top-left of the map north of Alaska by centering that corner of the map on your device? 

Offhand, there is nothing out of the box that would allow this. This is the same with MapKit. However, ours is open source, so I'm sure it can be done :smiley: 

Do you definitely need an interactive map? i.e. could this maybe be accomplished at a certain zoom level using the Mapbox static API and a custom `UIScrollView` setup? Can you elaborate on how the user gets to this point and what they do after?

**Comment by ghost** [comment]
That's correct, we need the user to be able to select, for example, the top-left corner of our map by centering that corner inside the device screen.

Our use case is a bit different to the normal ones. We don't display geospatial data but merely hallplans and seating maps for events. Zooming and scrolling is still needed for the User Experience though.

In this exact user story, the user selects an amount of seats and then a hallplan is shown where the user may select those seats. Since tapping is unprecise on size-changing objects we decided on a crosshair for selection (picking). The use case ends when the user confirms his selection of seats via a confirmation button.

We already modified a library on android to do exactly this. In that case we just changed some parameters on the scrolling behavior. The scrolling was limited to the map edges before but we were able to simply add half a screen size to those scrolling limits to allow for scrolling beyond the map edges.

We are looking for something similar with the Mapbox SDK. Any pointers would be greatly appreciated.

**Comment by incanus** [comment]
> We don't display geospatial data but merely hallplans and seating maps for events. Zooming and scrolling is still needed for the User Experience though.

I'm almost thinking a custom implementation similar to this library using `UIScrollView` & `CATiledLayer` might work better for you. Do you really need all of the feature of this SDK? When it comes to a hall plan, I'm guessing you don't even need the projection conversion features here because your reference system is linear.

**Comment by ghost** [comment]
That's certainly true. We don't need most of the features. Especially the projection part is making things worse for us since, as you already guessed, our reference system is linear.
We were hoping on being able to disable parts of the functionality we don't need. Also we didn't want to rewrite that whole UIScrollView & CATiledLayer mechanism when it is already implemented in a working manner here. Of course it may eventually prove more feasible in the end, if we should come to the conclusion that modifying Mapbox is way more complicated.
But since we are working on a deadline I'd prefer the solution requiring less coding time.
Also the last hurdle is the ability mentioned above, being able to scroll beyond the map boundaries. That doesn't seem to be out of scope. But we need some hints as where to make changes in the library.

**Comment by incanus** [comment]
Well, if you are going to stick with our tools, here's an idea. 
- Pick a coverage area that's not as large as the whole world that you will map to. 
- Create a custom `RMTileSource` to supply your tiles to that coverage area. 
- Have it supply blank or white tiles or something like that outside your coverage area. 

Then, you should be able to scroll to areas which show the blank/white tiles on screen, which means your true coverage area can still be centered. 

I'm going to close this now as it's rather off-topic for our toolset. Good luck. I'd recommend maybe checking out the WWDC sessions on scroll views (particularly 2011 & 2012) as they cover zooming of content views in the event that you need to build this up from scratch.

**Comment by ghost** [comment]
Thank you for your time. Your help is appreciated.
